tests/Unit/Evolution/Systems/ScalarTensor/Test_TimeDerivativeTerms.cpp

```diff
@@ -0,0 +1,259 @@
+                                   std::decay_t<decltype(t)>>) {
+            return *t;
+          } else {
+            return t;
+          }
+        }(tuples::get<ArgTags>(time_derivative_args))...);
+  }
+};
+
+} // namespace
+
+SPECTRE_TEST_CASE(
+    "Unit.Evolution.Systems.ScalarTensor.TimeDerivativeTerms",
+    "[Unit][Evolution]") {
+  using gh_variables_tags =
+      typename gh::System<3_st>::variables_tag::tags_list;
+  using scalar_variables_tags =
+      typename CurvedScalarWave::System<3_st>::variables_tag::tags_list;
+
+  using gh_dt_variables_tags =
+      ScalarTensor::TimeDerivativeTerms::gh_dt_tags;
+  using scalar_dt_variables_tags =
+      ScalarTensor::TimeDerivativeTerms::scalar_dt_tags;
+  using dt_variables_type = Variables<
+      tmpl::append<gh_dt_variables_tags, scalar_dt_variables_tags>>;
+
+  using gh_flux_tags = tmpl::list<>;
+  using scalar_flux_tags =
+      ScalarTensor::TimeDerivativeTerms::scalar_flux_tags;
+
+  using flux_variables_type =
+      Variables<tmpl::append<gh_flux_tags, scalar_flux_tags>>;
+
+  using gh_temp_tags = ScalarTensor::TimeDerivativeTerms::gh_temp_tags;
+  using scalar_temp_tags =
+      ScalarTensor::TimeDerivativeTerms::scalar_temp_tags;
+  using temp_variables_type = Variables<
+      typename ScalarTensor::TimeDerivativeTerms::temporary_tags>;
+
+  using gh_gradient_tags = tmpl::transform<
+      ScalarTensor::TimeDerivativeTerms::gh_gradient_tags,
+      tmpl::bind<::Tags::deriv, tmpl::_1, tmpl::pin<tmpl::size_t<3_st>>,
+                 tmpl::pin<Frame::Inertial>>>;
+  using scalar_gradient_tags = tmpl::transform<
+      ScalarTensor::TimeDerivativeTerms::scalar_gradient_tags,
+      tmpl::bind<::Tags::deriv, tmpl::_1, tmpl::pin<tmpl::size_t<3_st>>,
+                 tmpl::pin<Frame::Inertial>>>;
+  using gradient_variables_type =
+      Variables<tmpl::append<gh_gradient_tags, scalar_gradient_tags>>;
+
+  using gh_arg_tags =
+      ScalarTensor::TimeDerivativeTerms::gh_arg_tags;
+  using scalar_arg_tags =
+      ScalarTensor::TimeDerivativeTerms::scalar_arg_tags;
+  using arg_variables_type = tuples::tagged_tuple_from_typelist<
+      tmpl::append<gh_arg_tags, scalar_arg_tags,
+                   tmpl::list<ScalarTensor::Tags::ScalarSource>>>;
+
+  const size_t element_size = 10_st;
+  MAKE_GENERATOR(gen);
+  std::uniform_real_distribution<> dist(0.1, 1.0);
+
+  dt_variables_type expected_dt_variables{element_size};
+  dt_variables_type dt_variables{element_size};
+
+  flux_variables_type expected_flux_variables{element_size};
+  flux_variables_type flux_variables{element_size};
+
+  temp_variables_type expected_temp_variables{element_size};
+  temp_variables_type temp_variables{element_size};
+
+  const auto gradient_variables =
+      make_with_random_values<gradient_variables_type>(
+          make_not_null(&gen), make_not_null(&dist), DataVector{element_size});
+
+  arg_variables_type arg_variables;
+  tmpl::for_each<tmpl::append<
```

Comment:
It' would be a lot more readable to just remove the `MeshVelocity` from the tag list and create it separately so you don't need all of this.